deploy/charts/kubernetes-cluster-prep/templates/golden_configmap.yaml

```diff
@@ -0,0 +1,42 @@
+{{- if .Values.authnK8s.configMap.create -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+    name: {{ .Values.authnK8s.configMap.name }}
+    labels:
+      release: {{ .Release.Name }}
+      heritage: {{ .Release.Service }}
+      app.kubernetes.io/name: golden-configmap
+      app.kubernetes.io/component: "conjur-config"
+      app.kubernetes.io/instance: {{ .Release.Namespace }}
+      app.kubernetes.io/part-of: "authn-k8s-cluster-config"
+      helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+data:
+    # authn-k8s Configuration 
+    authnK8sAuthenticatorID: {{ required "A valid authnK8s.authenticatorID is required!" .Values.authnK8s.authenticatorID }}
+    {{- if eq .Values.authnK8s.clusterRole.create true }}
+    authnK8sClusterRole: {{ .Values.authnK8s.clusterRole.name | default "authn-k8s-clusterrole" }}
```

Comment:
If we set the values for `authnk8s.clusterRole.name` to "authn-k8s-clusterrole" in the `values.yml` file, do we need to check for a default value here? I guess my concern is if we change the default value, we have to go through and make sure each default check in each template uses the same value, whereas the `values.yml` file can act as the "defaults" implicitly. What do you think @diverdane ?

Reply:
Additionally, our schema validation makes sure that the value for `clusterRole.name` is a value with more than one characters, so it's not possible for the user to pass in a null or empty value without getting an error message, which I think would be the correct response, rather than proceeding with a default value that's hidden from the user.

Reply:
So if it's empty, would it error before setting the default? So in that case there is no reason to set it as you mention.

Reply:
The reason this `default` shows up in the chart is because we want this functionality:
- If the user is "bringing their own" ClusterRole, then `authnK8s.clusterRole.create` is set to `false`. In this case, they need to tell us the ClusterRole name, so `authnK8s.clusterRole.name` is required.
- If the user is having the chart create a ClusterRole for them, then `authnK8s.clusterRole.create` is set to `true`. In this case, we can create a ClusterRole with a default name of `authn-k8s-clusterrole`, or they can tell us what to name it.

Checking this in schema would be a bit tricky.